모집공고 에디터 추가 작업

sanoopark opened the pull request:
## 변경사항

![screenshot](https://github.com/mash-up-kr/mash-up-admin-fe/assets/81365896/1f9a7a6f-ed80-431c-bd16-3d4ca4b01745)

- 선택한 플랫폼에 따라 서버 데이터를 보여주도록 수정
- 조회, 수정 성공 / 실패 UI 추가
- API 연동 중 HTML Entity 이슈 수정
- 레이아웃 및 스타일 변경
- i18n 보완
- 기존 LNB 연동

위 내용을 포함해 전반적인 리팩토링도 진행했습니다

### 작업 유형

<!--  작업 유형에 맞는 리스트만 제외하고 지워주시면 됩니다 :) 해당 주석은 지우지 않아도 돼요!-->

- 신규 기능 추가
- 리팩토링
- 버그 수정

### 체크리스트

- [x] Merge 할 브랜치가 올바른가?
- [x] [코딩컨벤션](https://github.com/mash-up-kr/mash-up-recruit-fe/wiki/Coding-Convention)을 준수하였는가?
- [x] 해당 PR과 관련없는 변경사항이 없는가? (만약 있다면 제목이나 변경사항에 기술하여 주세요.)
- [ ] 실행시 console 창에 에러나 경고가 없는것을 확인하였는가? (개발에 필요하여 고의적으로 남겨둔것 제외)

swprk committed:
Feat: 에디터와 서버 데이터를 연동하다

swprk committed:
Fix: 서버 데이터의 HTML 엔티티를 수정하다

swprk committed:
Feat: 에디터 스타일을 수정하다

swprk committed:
Feat: LNB에 모집공고를 연결하다

swprk committed:
Feat: 에디터 레이아웃을 수정하다

swprk committed:
Feat: i18n 번역을 추가하다

Baek2back commented on src/components/common/Editor/tools.ts:
```diff
@@ -0,0 +1,55 @@
+   * @type {EditorConfig.tools}
+   */
+  header: {
+    class: Header,
```
@editorjs/* 에서 impor한 클래스를 직접 주입하게 되면 커스텀에 제약이 따르니

미리 한번 래핑해서 주입해주는 것도 좋을 것 같아요!

```typescript
import Header from '@editorjs/header';

class CMSHeader extends Header { ... }

export default {
  header: {
    class: CMSHeader,
    ...
  }
}
```

현재 API 응답을 보니 다음과 같은 형태로 내려오는 듯 한데, 추후에 커스텀이 가능하다면 data 객체 내에 className 프로퍼티를 추가하는 등의 형태로 정의할 수 있다면 recruit <-> admin 간에 공유해야 할 CSS 클래스 명을 더 직관적으로 가져갈 수 있을 것 같습니다.

https://github.com/editor-js/header/tree/master/src

우선은 현재 올려주신 작업 내용 기준으로 맞춰보고, 개선해보는 방향으로 진행해보시죠 👍

```json
[
  {
    "id":"0x4RZm_hm4",
    "type":"header",
    "data": {
      "text": "Product Design&amp;nbsp;팀 소개",
      "level":3,
      // ex. "className": "cms-header-3"
     },
    "tunes": { "textVariant": ""} 
  }
]
```

Baek2back reviewed (approved):
👍

swprk committed:
Refactor: 에디터 스타일 모듈을 분리하다

swprk committed:
Feat: Parser 미동작 플러그인을 제거하다

- `editorjs-html`에서 기본으로 지원하지 않는 플러그인을 1차 배포에서 제외

swprk committed:
Fix: 스타일로 인한 버그를 수정하다

- toolbar width로 인한 클릭 이벤트 미발생 수정
- header placeholder 클릭 커서 미발생 수정
- 기타 스타일 조정

swprk committed:
Feat: Tab 입력시 에디터 밖으로 Tabindex가 나가는 현상 수정
